File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        var stack = new Stack<Character>();
        var closeToOpen = new HashMap<Character, Character>();

        closeToOpen.put(')', '(');
        closeToOpen.put(']', '[');
        closeToOpen.put('}', '{');

        for (char c : s.toCharArray())
        {
            if (closeToOpen.containsKey(c))
            {
                if (!stack.isEmpty() && stack.peek() == closeToOpen.get(c))
                {
                    stack.pop();
                } else 
                {
                    return false;
                }
            } else
            {
                stack.push(c);
            }
        }

        return stack.isEmpty();
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for t in tokens:
            
            if t in "+":
                stack.append(stack.pop() + stack.pop())
            elif t in "-":
                a, b = stack.pop(), stack.pop()
                stack.append(b - a)
            elif t in "*":
                stack.append(stack.pop() * stack.pop())
            elif t in "/":
                a, b = stack.pop(), stack.pop()
                stack.append(int(float(b) / a))
            else:
                op = int(t)
                stack.append(op)
        
        return stack[0]